doc/man3/BIO_sendmmsg.pod

```diff
@@ -0,0 +1,160 @@
+was the loopback address or the IP address of a local non-loopback interface.
+This is an OS-specific limitation.
+
+The B<stride> argument must be set to B<sizeof(BIO_MSG)>. This argument
+facilitates backwards compatibility if flags are added to B<BIO_MSG>. Callers
+must zero-initialize B<BIO_MSG>.
+
+B<num_msg> should be sent to the maximum number of messages to send or receive,
+which is also the length of the array pointed to by B<msg>.
+
+Unlike most BIO functions, these functions explicitly support multi-threaded
+use. Multiple concurrent writers and multiple concurrent readers are permitted
+in any combination. As such, these functions do not clear, set, or otherwise
+modify BIO retry flags. The return value must be used to determine whether an
+operation should be retried; see below.
+
+BIO_dgram_set_local_addr_enable() and BIO_dgram_get_local_addr_enable() control
+whether local address support is enabled. To enable local address support, call
+BIO_dgram_set_local_addr_enable() with an argument of 1. The call will fail if
+local address support is not available for the platform.
+BIO_dgram_get_local_addr_enable() retrieves the value set by
+BIO_dgram_set_local_addr_enable().
+
+BIO_dgram_get_local_addr_cap() determines if the B<BIO> is capable of supporting
+local addresses.
+
+BIO_IS_ERRNO() and BIO_UNPACK_ERRNO() are used with the return values of
+BIO_sendmmsg() and BIO_recvmmsg(); see below.
+
+=head1 RETURN VALUES
+
+BIO_sendmmsg() and BIO_recvmmsg() the number of messages sent or received. Where
+a positive value n is returned, all entries in the B<BIO_MSG> array from 0
+through n-1 inclusive have their B<data_len> and B<flags> fields updated with
+the results of the operation on that message. If the call was to recvmmsg() and
+the B<peer> or B<local> fields of that message are non-NULL, the B<BIO_ADDR>
+structures they point to are written with the relevant address.
+
+If an error occurs before any message has been sent or received, a negative
+value is returned. -2 is returned if the BIO_sendmmsg() or BIO_recvmmsg() method
+is not supported by the BIO; -3 is returned if a local has been specified or
+requested and local address support is not available or not enabled; otherwise,
+a packed system error code is returned. The macro BIO_IS_ERRNO() evaluates to 1
+if a BIO_sendmmsg() or BIO_recvmmsg() return value represents a system error
+code. BIO_UNPACK_ERRNO() should then be used to obtain the system error code
+(EWOULDBLOCK, etc.) from the return value.
+
+If an error occurs after one or more messages have been sent or received that
+prevents sending or receiving further messages, the number of messages
+successfully sent or received is returned; the caller must learn of the error
+via a subsequent call to the same function.
+
+BIO_dgram_set_local_addr_enable() returns 1 if local address support was
+successfully enabled or disabled and 0 otherwise.
```

Comment:
Why do we need explicit `BIO_dgram_set_local_addr_enable()` call? Shouldn't the cap() call be sufficient? I.E., what are the use-cases where we are not interested in the local addr and does it bring enough performance difference to not have it always enabled where possible?

Reply:
We need this because local address support does need to be explicitly enabled by setting socket options. Doing this lazily/automatically would be a problem because we want to support multithreaded use of BIO_sendmmsg/BIO_recvmmsg without locking, so these functions should remain stateless.